cpp/benchmarks/CMakeLists.txt

```
@@ -52,6 +52,26 @@ function(ConfigureBench CMAKE_BENCH_NAME)
     )
 endfunction(ConfigureBench)
 
+# This function takes in a benchmark name and benchmark source for nvbench benchmarks and handles
+# setting all of the associated properties and linking to build the benchmark
+function(ConfigureNVBench CMAKE_BENCH_NAME)
+  add_executable(${CMAKE_BENCH_NAME} ${ARGN})
+  set_target_properties(
+    ${CMAKE_BENCH_NAME}
+    PROPERTIES RUNTIME_OUTPUT_DIRECTORY "$<BUILD_INTERFACE:${CUSPATIAL_BINARY_DIR}/benchmarks>"
+               INSTALL_RPATH "\$ORIGIN/../../../lib"
+  )
+  target_link_libraries(
+    ${CMAKE_BENCH_NAME} PRIVATE cuspatial_benchmark_common nvbench::main
+  )
+  install(
+    TARGETS ${CMAKE_BENCH_NAME}
+    COMPONENT testing
+    DESTINATION bin/benchmarks/libcuspatial
+    EXCLUDE_FROM_ALL
+  )
+endfunction(ConfigureNVBench)
```

Modern CMake style is not to specify this since it's redundant and just another source of potential error. I would change `ConfigureBench` above too.
```suggestion
endfunction()
```